File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOptArrayOptUUIDValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.OptArrayOptUUIDInValue.OptArrayOptUUIDValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOptArrayOptUUIDValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestOptArrayOptUUIDValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                java.util.UUID[].class,
                OptArrayOptUUIDValue.class.getDeclaredField("optArrayOptUUID").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                java.util.UUID[].class,
                OptArrayOptUUIDValue.class.getMethod("getOptArrayOptUUID").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                OptArrayOptUUIDValue.class,
                OptArrayOptUUIDValue.class.getMethod("setOptArrayOptUUID", java.util.UUID[].class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertNull(new OptArrayOptUUIDValue().getOptArrayOptUUID());
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOptArrayOptUUIDValue.scala
package com.dslplatform.ocd.values

import OptArrayOptUUIDInValue.OptArrayOptUUIDValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOptArrayOptUUIDValue extends SpecExtensions {
  def `Test property field type: Option[Array[Option[java.util.UUID]]]`() =
    checkType(
      typeOf[Option[Array[Option[java.util.UUID]]]]
    , typeOf[OptArrayOptUUIDValue].member("optArrayOptUUID": TermName).asMethod.returnType
    )

  def `Test property default value: None`() {
    assert(OptArrayOptUUIDValue().optArrayOptUUID === None)
  }
}
